client/hardening.ts

```diff
@@ -39,10 +39,26 @@ export async function fetchWithCSRF (url: string, requestInit: RequestInit): Pro
 	if (Array.isArray(tokenRaw)) token = tokenRaw[tokenRaw.length - 1];
 	else token = tokenRaw;
 	const bodyWithCSRF = { ...body, __CSRFToken: token };
```

Comment:
Also, more of an error in the original code, but body should be parsed before being combined with the CSRF token, otherwise it gets spread as a JSON string instead of a object.

Reply:
So a `...JSON.parse(body)` instead of `...body?`

Reply:
Yes.